### ISSUE TITLE
Can't properly initialize the pinning instance

### ISSUE THREAD

**Issue by rafaelkamimura** [frw/react-native-ssl-public-key-pinning#5]
Hi! So, I've been trying to implement this lib to my react-native project, but it doesn't seem to correctly pin the certificate. 
Problem: I need to make HTTPS requests on a server that uses Wild Card certificates, but the only other workaround, was the react-native-ssl-pinning, and it works just fine. But since I need to make many API calls to different endpoints, some of which require auth headers and such, and didn't want to change the whole Axios Instances.

The actual question is, where should the "initializeSslPinning" be called? Have tried several ways, but still getting the same old error (the following only happens on Android Devices):
```
"java.security.cert.CertPathValidatorException: Trust anchor for certification path not found."
```

Kind regards,
Rafael.

**Comment by frw** [comment]
Hey @rafaelkamimura,

> where should the "initializeSslPinning" be called?

`initializeSslPinning` should be called as early as possible, before your ever perform a network request. For example, you can call it within your app entrypoint (index.js), or you can call it right before you perform the network request itself.

> `"java.security.cert.CertPathValidatorException: Trust anchor for certification path not found."`

With regards to this error, are you only seeing this after calling `initializeSslPinning`? Does it not show up if you don't call `initializeSslPinning`? What Android version and React Native version are you using? If the error shows up regardless of whether `initializeSslPinning` is called, then perhaps you might have something [misconfigured on your server](https://stackoverflow.com/questions/6825226/trust-anchor-not-found-for-android-ssl-connection).

If you're willing to share the pinning configuration you're using, I could perhaps help to debug further.

**Comment by rafaelkamimura** [comment]
Thanks for the reply @frw ,
The thing is, i'm currently trying to create an axios instance as it follows:
```
import axios from 'axios';
import { initializeSslPinning } from 'react-native-ssl-public-key-pinning';

await initializeSslPinning({
    'anater.com': {
      includeSubdomains: true,
      publicKeyHashes: [
        'jgxahUXUF3a1cQ0QgIG6jAD8RxrPY8nX5LmOX068U1k=',
        '47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU='
      ],
    },
  });
  

const anaterApi = axios.create({ 
    baseURL: 'https://sga.anater.org',
    timeout: 10000,
});
export default anaterApi;
```
I've even tried to call `initializeSslPinning` on my Network Provider, but I might have misunderstood how it would go. 
The trust-anchor error happens for all requests whilst using axios for this domain due to its Wildcard SSL. The only way I'm being able to fetch data from the specified domain, is using the `react-native-ssl-pinning` fetch. We've even configured a temporary server with no CAs at all just for its initial release, but now we do need to change to the specified domain above for security reasons.
TLDR: I needed to be able to create a new axios instance to be reusable throughout the app itself, how wrong am I doing it?

Kind regards,
Rafael.

**Comment by frw** [comment]
Hey @rafaelkamimura,

I took a look at the domain you were trying to fetch, and it seems like the problem is indeed due to a misconfiguration of the server you were trying to connect to rather than due to the certificate being a wildcard certificate. Specifically, it does not include the full certificate chain from Let's Encrypt, which mean Android was unable to fully verify the entire chain from a trusted CA. More info about this here: https://stackoverflow.com/questions/6825226/trust-anchor-not-found-for-android-ssl-connection
If you input your domain on [SSL Lab](https://www.ssllabs.com/ssltest/index.html), it'll show this notice:
![image](https://user-images.githubusercontent.com/1888212/231645470-2b83aaab-82fc-45d5-8459-e842607a0571.png)


To fix this, you will need to make sure you're serving the full certificate chain. Let's Encrypt usually provides "fullchain.pem" during setup, which includes all of the certs from the leaf certificate all the way to the root CA certificate. 

Unfortunately, the pinning this library does only adds additional constraints on which certificates are accepted and it does not replace certificate authority requirements. I'm guessing you were using certificate pinning rather than public key pinning with  react-native-ssl-pinning, and I believe the reason why that worked is because react-native-ssl-pinning uses a custom client that overrides the Android trust store when performing certificate pinning.

**Comment by frw** [comment]
Also, as a side note, if reconfiguring your server is for some reason not possible, you could try adding a custom trust anchor for your Android app. Instructions to do this are available here: https://developer.android.com/training/articles/security-config#TrustingAdditionalCas

Do note that this is more of a workaround though, and will negate a lot of the flexibility benefits provided by this library. I would still recommend you get the server configured correctly to provide the full certificate chain.

**Comment by frw** [comment]
Closing this issue as it's unrelated to the library itself. Let me know if you have any further questions!
